### PR TITLE
ClassPropertyAssignToConstructorPromotionRector must skip callable pr…

### PR DESCRIPTION
**Pull request by tugmaks**
fixes https://github.com/rectorphp/rector/issues/6535

### CODE REVIEW

File: rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php [hunk 1]
```diff
@@ -161,4 +158,21 @@ private function decorateParamWithPropertyPhpDocInfo(Property $property, Param $
...
+            return true;
+        }
+
+        if ($param->type instanceof NullableType && $this->isName($param->type->type, 'callable')) {
+            return true;
+        }
+
+        if ($param->type instanceof Identifier && $this->isName($param->type, 'callable')) {
+            return true;
+        }
```
Comment (samsonasik) [inline rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php]: should can be simplified with: 

```suggestion
        $type = $param->type instanceof NullableType
            ? $param->type->type
            : $param->type;

        if ($type instanceof Identifier && $this->isName($type, 'callable')) {
            return true;
        }
```
to ensure the passed `$type` in `isName()` is always an identifier.

Comment (tugmaks) [reply rules/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector.php]: done

**Review by samsonasik — APPROVED** [review]
Thank you, the test error at `sabbelasichon/typo3-rector` seems unrelated.

@sabbelasichon could you check the test error at `sabbelasichon/typo3-rector`

https://github.com/rectorphp/rector-src/pull/338/checks?check_run_id=2949192987#step:5:24

Thank you.

**Comment by samsonasik** [comment]
@sabbelasichon I create PR https://github.com/sabbelasichon/typo3-rector/pull/2486 for that for possible solution.

**Comment by samsonasik** [comment]
test workflow restarted

**Review by samsonasik — APPROVED** [review]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**